eegnb/cli/introprompt.py

```diff
@@ -40,57 +40,61 @@ def intro_prompt():
     print(f"Selected board {boards[board_idx]} \n")
 
     # Handles wifi shield connectivity selection if an OpenBCI board is being used
     if board_selection in ['cyton', 'cyton_daisy', 'ganglion']:
-        # if the ganglion is being used, will also need the MAC address
-        if board_selection == 'ganglion':
-            print("Please enter the Ganglions MAC address:\n")
-            mac_address = input("MAC address:")
 
         # determine whether board is connected via Wifi or BLE
         print("Please select your connection method:\n"
               "[0] usb dongle \n"
               "[1] wifi shield \n")
-        connect_idx = input("Enter connection method:")
+        connect_idx = int(input("Enter connection method: "))
 
         # add "_wifi" suffix to the end of the board name for brainflow
         if connect_idx == 1:
             board_selection = board_selection + "_wifi"
+            ip_address = input("\nEnter Ganglion+WiFi IP Address: ")
+        else:
+            # if the ganglion is being used, you can enter optional Ganglion mac address
+            if board_selection == 'ganglion':
+                ganglion_mac_address = input("\nGanglion MAC Address (Press Enter to Autoscan): ")
 
     # Experiment selection
-    print("Please select which experiment you would like to run: \n"
+    print("\nPlease select which experiment you would like to run: \n"
           "[0] visual n170 \n"
           "[1] visual p300 \n"
           "[2] ssvep \n")
 
-    exp_idx = int(input('Enter Experiment Selection:'))
+    exp_idx = int(input('Enter Experiment Selection: '))
     exp_selection = experiments[exp_idx]
     print(f"Selected experiment {exp_selection} \n")
 
     # record duration
     print("Now, enter the duration of the recording (in seconds). \n")
-    duration = int(input("Enter duration:"))
+    duration = int(input("Enter duration: "))
 
     # Subject ID specification
-    print("Next, enter the ID# of the subject you are recording data from. \n")
-    subj_id = int(input("Enter subject ID#:"))
+    print("\nNext, enter the ID# of the subject you are recording data from. \n")
+    subj_id = int(input("Enter subject ID#: "))
 
     # Session ID specification
-    print("Next, enter the session number you are recording for. \n")
-    session_nb = int(input("Enter session #:"))
+    print("\nNext, enter the session number you are recording for. \n")
+    session_nb = int(input("Enter session #: "))
 
     # start the EEG device
-    if board_selection == 'ganglion':
-        # if the ganglion is chosen a MAC address should also be proviced
-        eeg_device = EEG(device=board_selection, mac_addr=mac_address)
+    if board_selection.startswith('ganglion'):
+        if board_selection == 'ganglion_wifi':
+            eeg_device = EEG(device=board_selection, ip_addr=ip_address)
+        else: 
+            eeg_device = EEG(device=board_selection, mac_addr=ganglion_mac_address)
```

Comment:
The ganglion really isn't special here (afaik). Looks like it's called the same anyway.

```suggestion
    if not board_selection.endswith('_wifi'):
        eeg_device = EEG(device=board_selection, mac_addr=mac_address)
```

Reply:
```
# start the EEG device
    if board_selection.startswith('ganglion'):
        if board_selection == 'ganglion_wifi':
            eeg_device = EEG(device=board_selection, ip_addr=ip_address)
        else: 
            eeg_device = EEG(device=board_selection, mac_addr=ganglion_mac_address)
    else:
        eeg_device = EEG(device=board_selection) <---Fixed this
```

Reply:
See: https://github.com/NeuroTechX/eeg-notebooks/pull/41#discussion_r535452277